community/gill-supabase-web3/src/app/api/auth/wallet/route.ts

```diff
@@ -0,0 +1,58 @@
+import { NextRequest, NextResponse } from 'next/server'
+import { prisma } from '@/lib/prisma'
+
+export async function POST(request: NextRequest) {
+  try {
+    const { walletAddress, supabaseUserId } = await request.json()
+
+    if (!walletAddress) {
+      return NextResponse.json({ error: 'Wallet address is required' }, { status: 400 })
+    }
+
+    // First, check if user already exists in the database
+    const existingUser = await prisma.user.findUnique({
```

Comment:
do we need the user since Supabase supports already singup methods with of web3 automatically?

https://supabase.com/docs/guides/auth/auth-web3